### ISSUE TITLE
After several rounds, the loss function will become nan. 

### ISSUE THREAD

**Issue by zhrli** [ryanchankh/mcr2#10]
logdet value becomes nan, when I  used  the UCR dataset named FaceDetecion.
I chosed the BERT model to deal with the multiple timeseries data. 
I wonder why this happened?

**Comment by waterdisappear** [comment]
I think normalize is helpful.

```
    def forward(self, X, Y, num_classes=None):
        if num_classes is None:
            num_classes = Y.max() + 1
        X = F.normalize(X, dim=-1)
        W = X.T
        Pi = tf.label_to_membership(Y.numpy(), num_classes)
        Pi = torch.tensor(Pi, dtype=torch.float32).cuda()
```
